velox/exec/RowContainer.h

```diff
@@ -671,18 +671,11 @@ class RowContainer {
   void
   setProbedFlag(char** rows, int32_t numRows);
 
-  /// Returns true if 'row' at 'column' equals the value at 'index' in
-  /// 'decoded'. 'mayHaveNulls' specifies if nulls need to be checked. This is a
-  /// fast path for compare().
-  template <bool mayHaveNulls>
-  bool equals(
```

Comment:
Hello, for types like strings equals can be implemented a lot of more efficient than compare == 0, does velox and your PR account this for
1) varchar/varbinary
2) custom types

Reply:
I understand, thanks!

It's not related to your PR but maybe you know, why such design (with compare == 0 + flags) was chosen instead of separate equals method?

Reply:
For an example

```
  // If equalsOnly and maps have different sizes, we can bail fast.
  if (flags.equalsOnly && leftIndices.size() != rightIndices.size()) {
    int result = leftIndices.size() - rightIndices.size();
    return flags.ascending ? result : result * -1;
  }
```

here in case of equalsOnly we still tries to distinguish -1 and 1, although it doesn't really matter